mydemoproject/demoapp1/models/user.py

```diff
@@ -23,13 +22,17 @@ def __str__(self):
     def by_email(cls, email):
         return cls.filter_objects(email=email).exists()
 
+    @classmethod
+    def by_phone(cls, phone_number):
+        return cls.filter_objects(phone_number=phone_number).exists()
+
     @classmethod
     def add_user(cls, email, phone_number, password, **extra_fields):
         if cls.by_email(email):
-            raise ValidationError('User already exists')
+            return False, "User already exists"
+        if cls.by_phone(phone_number):
+            return False, "Phone-Number already registered"
 
-        # We have to put else otherwise it will give warning
-        # of code unreachable
         else:
```

Comment:
No need for an else here as I have already told we need reduce the if-else chaining.